kong/plugins/key-auth/handler.lua

```diff
@@ -105,6 +106,13 @@ local function unauthorized(message, www_auth_content)
   return { status = 401, message = message, headers = { ["WWW-Authenticate"] = www_auth_content } }
 end
 
+local function remove_query_key(raw_query, key)
+  local pattern = key .. "=[^&]*&?"
```

Comment:
@battlebyte - chatting with @bungle, we questioned what would happen if the query arg is url-encoded. Can you think about this for a bit and verify?

Reply:
That is a good point and I thought about that. According to the plugin documentation, 

> The key names may only contain [a-z], [A-Z], [0-9], [_] underscore, and [-] hyphen.

which looks to me those were carefully chosen to avoid url-encoding. I checked the url encoding functions of several programming languages and they do not encode those characters. For example, those won't be encoded in

> [ngx.escape_uri](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxescape_uri)

But certainly, it still could be that it is url-encoded. I could use 

> [ngx.unescape_uri](https://github.com/openresty/lua-nginx-module?tab=readme-ov-file#ngxunescape_uri)

 to make sure we are dealing with a url-decoded key name before applying the substitution. 

 What do you think?

Reply:
What about the separator `&` - that we are using in the regex?

Reply:
Any concerns here @bungle?

Reply:
Latest commit uses ngx.unescape_uri to remove the key only on uri decoded query strings.